chore: kubernetes single-stop page for reference arch (#5034)
camunda/camunda-docs: docs/self-managed/reference-architecture/kubernetes.md

## Key features

- **Scalability & High Availability**: Camunda 8 can scale up and down depending on your demand, and makes use of a highly available setup.
- **Fault Tolerance & Resilience**: Scale Camunda 8 across zones or across regions to provide greater fault tolerance and uptime for your workflows.

Although Kubernetes has a steeper learning curve, once properly configured, it provides significant advantages, including self-healing capabilities that ensure application resilience by automatically restarting failed containers and rescheduling workloads.

Additionally, Kubernetes benefits from a vast ecosystem of **[Cloud Native Computing Foundation (CNCF) projects](https://www.cncf.io/)**, offering seamless integration with monitoring, observability, logging, and security solutions. Tools like Prometheus and Grafana allow for comprehensive monitoring, while service meshes such as Istio enhance traffic management and security.

By leveraging this ecosystem, organizations can extend Kubernetes functionality to meet their specific operational needs, enabling greater automation, scalability, and visibility across their infrastructure.

> ## Reference implementations

Reviewer: I am wondering if this section would be better incorporated elsewhere. Does it make sense to provide these two (and only two) references up front?

Could we to the overview for each cloud provider within the cloud-specific section at the end?

Author: I remember for manual I had them at the bottom in the beginning and then they got moved up.

The amount will grow over time, e.g. Azure in the workings atm.

The problem I see with moving them to the cloud-specific part is that you don't immediately see them in the table of contents anymore and people may miss it.

This section includes deployment reference architectures for manual setups:

- [Amazon EKS single-region](/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md): a standard production setup.
- [Amazon EKS dual-region](/self-managed/setup/deploy/amazon/amazon-eks/dual-region.md): an advanced use case utilizing two regions.

For general deployment pitfalls, visit the [deployment troubleshooting guide](/self-managed/operational-guides/troubleshooting/troubleshooting.md).

## Considerations

- Most reference implementations primarily focus on the [Orchestration cluster](#components).

> - When scaling the web applications (Operate, Tasklist, Optimize), ensure that the importer/archiver runs only once within the whole cluster. See [high availability](#high-availability-ha) for for more information.

Reviewer: I am not sure this is needed. We haven't yet really defined Orchestration cluster, and don't describe why this might be something worth considering. The importer/archiver note is clear in its section as well.

Author: you mean removing considerations as a whole? Fine by me.

Zeebe Brokers also have a service but are not directly exposed externally; all requests must pass through the Zeebe Gateway, which is typically used for internal communication as well.

#### Web Modeler and Console

![Web Modeler and Console](./img/k8s-cluster-view-managing.jpg)

Web Modeler, Console, and Identity are deployed as **Deployments** since they are stateless, with data stored externally in a SQL database. This allows them to be easily scaled as needed.

The namespace has its own Ingress resource, as those are namespace-bound and not cluster-wide resources. This means you will have to use a different subdomain for each Ingress resource. For more details, refer to the [multi-namespace deployment guide](/self-managed/setup/guides/multi-namespace-deployment.md).

### High availability (HA)

:::caution Non-HA importer / archiver
The following concerns **Operate**, **Tasklist**, and **Optimize**:

When scaling from a single pod to multiple, ensure that the `importer / archiver` is enabled on only one pod and disabled on others. Enabling it on multiple pods will cause data inconsistencies. This limitation is known and will be addressed in future updates.

> Example configuration options are available for [Operate](/self-managed/operate-deployment/importer-and-archiver.md#configuration) and [Tasklist](/self-managed/tasklist-deployment/importer-and-archiver.md#configuration).

Reviewer: I couldn't find the configuration examples for Optimize. 😬 Not sure if that should be flagged elsewhere to add in the docs, or if the configuration is identical. We could just mention that if so.

The default node affinity settings prevent Zeebe brokers from being scheduled on the same node, requiring a minimum three-node setup. For more details on Kubernetes affinity rules, see the [official documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/).

To further enhance fault tolerance, it is recommended to distribute Zeebe brokers and other scaled applications across multiple availability zones using additional affinity rules, ensuring resilience in case of a zone failure.

### Components

A typical Camunda 8 deployment distinguishes between the **Orchestration cluster** and the **Web Modeler and Console**. These clusters should be separated by utilizing [Kubernetes namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/), though running within a single namespace is possible.

> A multi-namespace setup allows you to duplicate and run multiple Orchestration clusters based on different use cases and requirements, while Web Modeler and Console remains independent, and does not need to be scaled the same way. The same applies to Identity as a central entity coordinating access management.

Reviewer: This is a section I reworded pretty significantly - just wanted to call it out specifically for review!

### Infrastructure

We recommend using an officially [certified Kubernetes](https://www.cncf.io/training/certification/software-conformance/#benefits) distribution.

Camunda 8 is not tied to a specific Kubernetes version. To simplify deployment, we provide a [Helm chart](/self-managed/setup/install.md) for easy installation on Kubernetes. The latest Helm chart is typically compatible with Kubernetes' [official support cycle](https://kubernetes.io/releases/).

#### Minimum cluster requirements

The following are suggested minimum requirements. Sizing depends heavily on your specific use cases and workload. See [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) and [Zeebe resource planning](/self-managed/zeebe-deployment/operations/resource-planning.md) for additional resources, conduct benchmarking to determine your exact requirements.

- 4 Kubernetes Nodes
  - Modern CPU: 4 Cores
  - Memory: 16 GiB
- Persistent Volumes
  - 1,000 IOPS
  - 32 GiB size
  - _avoid burstable disk types_

#### Networking

> While networking is largely managed through services and Load Balancers, understanding port usage can be useful - especially if explicit whitelisting is required within a private network.

Reviewer: I wanted to rephrase this but wasn't entirely sure:
```suggestion
Networking is largely managed through services and Load Balancers. The following describes networking requirements, including port usage, which may be necessary if explicit whitelisting is required within a private network.
```

Author: Not 100% sure with the wording either, let's go with your version for now.

Author: backported it to 8.7 👍

- Stable, high-speed network connection
- Configured firewall rules to allow necessary traffic:
  - **80**: Web UI (Console, Identity, Keycloak, Operate, Optimize, Tasklist, Web Modeler)
  - **82**: Metrics endpoint (Identity)
  - **8080**: REST endpoint (Connectors, Keycloak, Zeebe Gateway)
  - **8071 / 8091**: Management endpoint (Web Modeler)
  - **8092**: Management endpoint (Optimize)
  - **9100**: Management endpoint (Console)
  - **9600**: Management endpoint (Operate, Tasklist, Zeebe Brokers, Zeebe Gateway)
  - **26500**: gRPC endpoint.
  - **26501**: Gateway-to-broker communication.
  - **26502**: Inter-broker communication.
- Load Balancer for distributing traffic and exposing Camunda 8 to users (if required)

The exposed Kubernetes service port does not always match the target port of the components. The correct target ports can be found in the rendered Helm chart, or in the configuration pages for each component.

> :::note Databases
> 
> Databases were not considered as those should be maintained outside and ports may differ from the upstream defaults.
> 
> The defaults for the databases are:
> 
> - **5432**: PostgreSQL
> - **9200 / 9300**: Elasticsearch / OpenSearch
> 
> :::

Reviewer: This note isn't terribly straightforward to me. Does this mean:
```suggestion
:::note Databases

Database ports are not included, as databases should be maintained outside Camunda, and ports may differ from the upstream defaults.

The defaults for the databases are:

- **5432**: PostgreSQL
- **9200 / 9300**: Elasticsearch/OpenSearch

:::
```

Author: backported it to 8.7 👍
